docs/sphinx/using/cudaq/platform.rst

```diff
@@ -45,9 +45,217 @@ Here is a simple example demonstrating this
+in order to compute the expectation value of that Hamiltonian given the quantum state prepared by the ansatz kernel. When the :code:`nvidia-mqpu` platform
+is selected, these circuits will be distributed across all available QPUs. The final expectation value result is computed from all QPU execution results.
+
+Parallel distribution mode
+++++++++++++++++++++++++++
+
+The CUDA Quantum :code:`nvidia-mqpu` platform supports two modes of parallel distribution of expectation value computation:
+
+* MPI: distribute the expectation value computations across available MPI ranks and GPUs for each Hamiltonian term.
+* Thread: distribute the expectation value computations among available GPUs via standard C++ threads (each thread handles one GPU).
+
+For instance, if all GPUs are available on a single node, thread-based parallel distribution 
+(:code:`cudaq::parallel::thread` in C++ or :code:`cudaq.parallel.thread` in Python, as shown in the above example) is sufficient. 
+On the other hand, if one wants to distribute the tasks across GPUs on multiple nodes, e.g., on a compute cluster, MPI distribution mode
+should be used.
+
+An example of MPI distribution mode usage is as follows:
+
+C++
+^^^
+
+.. code-block:: cpp 
+
+    #include "cudaq.h"
+
+    int main() {
+      cudaq::mpi::initialize();
+      using namespace cudaq::spin;
+      cudaq::spin_op h = 5.907 - 2.1433 * x(0) * x(1) - 2.1433 * y(0) * y(1) +
+                        .21829 * z(0) - 6.125 * z(1);
+
+      auto ansatz = [](double theta) __qpu__ {
+        cudaq::qubit q, r;
+        x(q);
+        ry(theta, r);
+        x<cudaq::ctrl>(r, q);
+      };
+
+      double result = cudaq::observe<cudaq::parallel::mpi>(ansatz, h, 0.59);
+      if (cudaq::mpi::rank() == 0)
+        printf("Expectation value: %lf\n", result);
+      cudaq::mpi::finalize();
+
+      return 0;
+    }
+
+.. code-block:: console 
+
+    nvq++ observe.cpp -target nvidia-mqpu
+    mpirun -np <N> a.out
+
+
+Python
+^^^^^^
+
+.. code:: python
+
+    import cudaq
+    from cudaq import spin
+
+    cudaq.mpi.initialize()
+    cudaq.set_target("nvidia-mqpu")
+
+    # Define spin ansatz.
+    kernel, theta = cudaq.make_kernel(float)
+    qreg = kernel.qalloc(2)
+    kernel.x(qreg[0])
+    kernel.ry(theta, qreg[1])
+    kernel.cx(qreg[1], qreg[0])
+    # Define spin Hamiltonian.
+    hamiltonian = 5.907 - 2.1433 * spin.x(0) * spin.x(1) - 2.1433 * spin.y(
+        0) * spin.y(1) + .21829 * spin.z(0) - 6.125 * spin.z(1)
+
+
+    exp_val = cudaq.observe(kernel, hamiltonian, 0.59, execution=cudaq.parallel.mpi).expectation_z()
+    if cudaq.mpi.rank() == 0:
+        print("Expectation value: ", exp_val)
+
+
+    cudaq.mpi.finalize()
+
+.. code-block:: console 
+
+    mpirun -np <N> python3 observe_mpi.py
+
+In the above examples, we specified the parallel distribution mode to :code:`mpi` and used CUDA Quantum MPI utility functions 
```

Comment:
nit: May want to remove first-person sorts of writing (we/I) and rephrase?

Reply:
I've rephrased the sentence to remove first-person pronouns.